10379.xml

```diff
@@ -0,0 +1,129 @@
+                <Name>Linked Sensor</Name>
+                <Operations>R</Operations>
+                <MultipleInstances>Single</MultipleInstances>
+                <Mandatory>Mandatory</Mandatory>
+                <Type>Objlnk</Type>
+                <RangeEnumeration></RangeEnumeration>
+                <Units></Units>
+                <Description>Link to the sensor object instance that this metadata relates to.
+                </Description>
+            </Item>
+            <Item ID="1">
+                <Name>Measurement Quality Reason Code</Name>
+                <Operations>R</Operations>
+                <MultipleInstances>Multiple</MultipleInstances>
+                <Mandatory>Optional</Mandatory>
+                <Type>Integer</Type>
+                <RangeEnumeration>0..1024</RangeEnumeration>
+                <Units></Units>
+                <Description>Measurement quality degradation reason reported by a smart sensor. 
+A degraded measurement has Measurement Quality Indicator value 1-3 or Measurement Quality Level less than 100.
+0: UNKNOWN Reason is unknown.
+1: UNDER_RANGE Measured value is under measurement range. 
+2: OVER_RANGE Measured value is over measurement range.
+3: NOISY Measured value seems to change too quickly. 
+4: STUCK_VALUE Measured value seems to change too slowly.
+5: STEP_TOO_HIGH Measured value has changed too much.
+6: CALIBRATION_EXPIRED Sensor calibration has expired.
+7: WARMING_UP Sensor is warming up.
+8: SERVICE_MODE Sensor is under service.
+9: HUMAN_INTERFERENCE Human presence near the sensor.
+10: NATURAL_INTERFERENCE Natural phenomenon or animal presence.
+11: CONTAMINATION Contamination of the sensor.
+12: HW_FAILURE Sensor hardware has failed.
+13: LACK_OF_DATA_SAMPLES Data sampling is lossy.
+14: OUT_OF_OPERATING_RANGE Environment conditions exceeded.
+15: AUTO_CALIBRATION Auto calibration in progress. 
+16-64: RESERVED_GENERIC_BLOCK_1 Reserved block 1.
+65-128: VENDOR_SPECIFIC_BLOCK_1 Vendor specific block #1.
+129-255: RESERVED_VENDOR_BLOCK_2 Reserved block 2.
+256-1024: VENDOR_SPECIFIC_BLOCK_2 Vendor specific block #2.
+                </Description>
+            </Item>
+            <Item ID="6042">
+                <Name>Measurement Quality Indicator</Name>
+                <Operations>R</Operations>
+                <MultipleInstances>Single</MultipleInstances>
+                <Mandatory>Optional</Mandatory>
+                <Type>Integer</Type>
+                <RangeEnumeration>0..23</RangeEnumeration>
+                <Units></Units>
+                <Description>Measurement quality indicator reported by a smart sensor.
+0: UNCHECKED No quality checks were done because they do not exist or can not be applied.
+1: REJECTED WITH CERTAINTY The measured value is invalid.
+2: REJECTED WITH PROBABILITY The measured value is likely invalid.
+3: ACCEPTED BUT SUSPICIOUS The measured value is likely OK.
+4: ACCEPTED The measured value is OK.
+5-15: Reserved for future extensions.
```

Comment:
Like you did in the VSMQR object, perhaps it's a good policy to give short names also to the ranges. For example `RESERVED` and `VENDOR_SPECIFIC`.

Reply:
That would break the validation unless we also change re-usable resource 6042 description. Should I include that into the same pr or should we handle it separately?

Reply:
I agree that it would be more consistent, but we would need to make changes in many objects using 6042, as I noted above. A new dedicated pr is my suggestion, if we decide to set this right.

Reply:
Good point; I didn't realize this change would have such a big impact. Let's park this change suggestion for the next round of updates that go across all / many IPSO objects.